packages/integration-tests/src/test-e2e/amplify_outputs_backwards_compatibility.test.ts

```diff
@@ -99,6 +97,64 @@ void describe('client config backwards compatibility', () => {
     });
   };
 
+  const assertGenerateClientConfigAPI = async (
+    type: 'baseline' | 'current'
+  ) => {
+    try {
+      assert.ok(
+        await generateClientConfig(branchBackendIdentifier, '1'),
+        `outputs v1 failed to be generated for an app created with ${type} library version`
+      );
+    } catch (e) {
+      throw new Error(
+        `outputs v1 failed to be generated for an app created with ${type} library version. Error: ${JSON.stringify(
+          e
+        )}`
+      );
```

Comment:
do we need these try-catch blocks ?

would `assert.doesNotReject()` work here ?

Reply:
it will, but the error messages were a bit cryptic. This way we can add more assertions on the result of the API as well in the future.